onenetbook/4/default.nix

```diff
@@ -0,0 +1,43 @@
+{ pkgs, config, ... }:
+
+{
+  imports = [
+    ../../common/cpu/intel
+    ../../common/pc/laptop
+    ../../common/pc/laptop/ssd
+  ];
+
+  # OneNetbook 4 has `GXTP7386:00 27C6:011A Stylus` exporting no buttons in 5.12
+  # and libinput does't consider it a tablet without them, but a touchscreen.
+  # This leads to real weird effects,
+  # starting from clicking as soon as the pen gets tracked.
+  # A kernel patch exists to resolve that, compiled as an out-of-tree module here
+  # to avoid recompiling the kernel for such a small change.
+  # `hid-multitouch-onenetbook4` is the fixed one, don't use `hid-multitouch`.
+  boot.blacklistedKernelModules = [ "hid-multitouch" ];
+  boot.extraModulePackages = [
+    (config.boot.kernelPackages.callPackage ./goodix-stylus-mastykin {})
+  ];
+
+  # has an accelerometer
+  hardware.sensor.iio.enable = true;
+  # accelerometer needs rotation
+  # (this at least gets normal/left-up/right-up/bottom-up right)
+  services.udev.extraHwdb = ''
+    acpi:BOSC0200:BOSC0200:*
+     ACCEL_MOUNT_MATRIX=0, 1, 0; 0, 0, 1; 1, 0, 0
+  '';
+  # but this, in turn, requires iio-sensor-proxy 3.0+
+  nixpkgs.overlays = [
+    (self: super: {
+      iio-sensor-proxy = super.iio-sensor-proxy.overrideAttrs (oa: rec {
+        version = "3.0";
+        src = pkgs.fetchgit {
+          url = "https://gitlab.freedesktop.org/hadess/iio-sensor-proxy";
+          rev = version;
+          sha256 = "sha256-ibxA0xVsm+wVYrlXVOfaNypmmIHykGtoHZHJNXf461k=";
+        };
+      });
```

Comment:
I'm pretty sure we don't want to include such updates in the nixos-hardware repo. At the very least not without *also* providing the proper update in Nixpkgs. I opened the upgrade here https://github.com/NixOS/nixpkgs/pull/125989 as @r-ryantm won't pick it up as it is.

Though we *might* want to include this for current-stable (and current-unstable)... Though most likely we want to check with `lib.versionOlder super.iio-sensor-proxy.version "3.0"` and conditionally add the update. This way further upgrades are not clobbered by this.

Furthermore, we should use `fetchFromGitLab` here. As it uses similar semantics to `fetchFromGitHub`, which in turn prevents loading the whole git history.

Reply:
Ack, will look into fetchGitLab and conditioning this tomorrow.

Reply:
I've slept over it and I agree that this repository should rather deal with configuration, not packaging. I think there's very little harm in conditioning the rotation on the software version + just suggesting how to update. PR updated (e138a81).

Plus, this isn't exactly wildly popular hardware, the amount of people inconvenienced with this is gonna be single digits at best, most likely zero. I think updating software through this repo still might make sense in some rare cases, but it should be reserved to smth like mitigating regressions from bad firmware updates, not something as puny as this case.